### DISCUSSION TITLE
[Feature] Show owner of picture in shared album

### DISCUSSION THREAD

**Discussion by Chacsam** [immich-app/immich#8779] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

When several people upload their pictures to a shared album, it would be nice to see who's the original author/owner of the picture.
It could be either:
- In the "info" section of the picture
- A small watermark at the bottom right of the picture (maybe on mouseover)

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by AlexanderBabel** [comment]
ente.io offers a similar feature. As soon as someone is uploading pictures anonymously to a shared photo album, they are asked to provide a name.

For me, this is a big use case when trying to share pictures between platforms. For instance, I have some people on an iCloud Shared Photo album that is backed up to immich. Then I want the pictures from people with an Android device as well on Immich and on iCloud. For that, I have to manually download them from the album and add them to iCloud. This can be a lot easier by seeing who was actually uploading the pictures in the first place.

**Comment by miguent** [comment]
Voting +1 for this feature.
I just added several users to my immich instance to share photos within our group of people who were on the same holiday.
I created an Album for sharing my photos with these guys and ask them to upload their photos as well and add them to this shared album. Would be nice to see which photo was uploaded by whom and maybe also have a filtering option for that.

**Comment by aviv926** [comment]
almost similar to https://github.com/immich-app/immich/discussions/7608

**Reply by AlexanderBabel** [reply]
I think that feature can be part of this feature request because it only implements the UI in the viewer for the other already existing partner sharing feature.

**Reply by miguent** [reply]
True. I would propose to show a small badge on the photo with the profile picture of the user who owns this photo. This I would say is sufficient only in case there are photos from multiple users in an album (or in the timeline via the partner share).

**Reply by lx05** [reply]
yeah this feature would be great, so I can just use 1 shared album for my vacation with friends but later on I can still see, which pictures were taken by me. Maybe even a filter option in the ui would be helpful. Just a small selection: "Show pictures taken by" where you can select and deselect.

**Comment by PeggyFree** [comment]
I confirm this is really essential when people share their images

**Comment by justincoffman** [comment]
Bumping this, would be really helpful to see who uploaded to the shared album. 

Cluster is another app that supports this feature for reference (I think you can also group by who uploaded).

**Reply by bo0tzz** [reply]
Bumping doesn't do anything except notify people unnecessarily. If you want to support the feature request, just upvote the original post please.

**Comment by ElioDiNino** [comment]
Super happy to see the show owner option added in v2.4.0 as part of #21171 for web, but would also love to see something similar for the mobile app!

**Reply by StevenBrehmer** [reply]
Agreed- would be very useful on mobile.

**Reply by TimberlAndRE-de** [reply]
Yes! This would be great, please add this feature to the mobile app too.

**Comment by ahmetem** [comment]
Agreed- would be very useful on mobile.

**Comment by jenda69** [comment]
Hm, I wonder, is there a way to filter photos in an album by a user?
I have shared album with my wife with photos of our child and if I receive a notification new photos were added, I have to go and find them manually mixed between photos from me.
Showing owner names helps a lot but is not perfect, especially because highlighting it using Ctrl+F doesn't really work because of progressive loading of the page.

**Comment by bo0tzz** [comment]
#29302
